lib/recommender_system/irt_recommender.rb

```diff
@@ -0,0 +1,193 @@
+module RecommenderSystem
+  class IrtRecommender < RecommenderSystem::Recommender
+    # Metoda vytvori list learning objektov s priradenymi hodnotami na zaklade pravdepodobnosti spravnej odpovede urcenej cez Item Response Theory
+    def get_list
+      los = self.learning_objects
+      list = Hash.new
+      user = User.find(self.user_id)
+      proficiency = user.proficiency
+      los.each do |lo|
+        difficulty = lo.irt_difficulty || 0.5
+        discrimination = lo.irt_discrimination || 0.5
+        # Vypocet pravdepodobnosti na zaklade 2P IRT
+        probability = (Math::E**(discrimination*(proficiency-difficulty))) / (1 + Math::E**(discrimination*(proficiency-difficulty)))
+        list[lo.id] = probability
+      end
+      list
+    end
+
+    # Opakuje sa o stvrtej rano, vid. config/schedule.rb
+    def self.update_table
+      require "rinruby"
+      # Predspracovanie dat
+      result_relations = Levels::Preproces.preproces_data
+      number_of_users = User.all.order("id DESC").first.id
+      number_of_los = LearningObject.all.order("id DESC").first.id
```

Comment:
Vrati `nil`, pretoze ten gem/defaultny scope tam automaticky pridava dalsiu podmienku:
```
 LearningObject.find(1)

SELECT  "learning_objects".* FROM "learning_objects" WHERE 

>>>> "learning_objects"."deleted_at" IS NULL <<<<

AND "learning_objects"."id" = $1 LIMIT 1
```

Cize treba to dole upravit, lebo `LearningObject.find(i).update!` spravi v takom pripade `nil.update!`.